src/main/java/ui/listpanel/ListPanel.java

```diff
@@ -279,14 +275,18 @@ private void setupKeyboardShortcuts() {
                 }
             }
             if (MANAGE_ASSIGNEES.match(event) && ui.getBrowserComponent().isCurrentUrlIssue()) {
-                ui.getBrowserComponent().switchToTab(DISCUSSION_TAB);
+                if (ui.getBrowserComponent().isCurrentUrlPR()) {
+                    ui.getBrowserComponent().openPrTab(getCurrentlySelectedIssue(), DISCUSSION_TAB);
+                }
                 ui.getBrowserComponent().manageAssignees(event.getCode().toString());
             }
             if (SHOW_MILESTONES.match(event)) {
                 if (KeyPress.isValidKeyCombination(GOTO_MODIFIER.getCode(), event.getCode())) {
                     ui.getBrowserComponent().showMilestones();
                 } else if (ui.getBrowserComponent().isCurrentUrlIssue()) {
-                    ui.getBrowserComponent().switchToTab(DISCUSSION_TAB);
+                    if (ui.getBrowserComponent().isCurrentUrlPR()) {
+                        ui.getBrowserComponent().openPrTab(getCurrentlySelectedIssue(), DISCUSSION_TAB);
```

Comment:
I shouldn't change `isPullRequestLoaded` because there are a few tests that specifically test for that behaviour.

Having a new check for whether the current url is in the PR page but not discussion tab also seems redundant, because `openPrTab` already handles the case where the url is in the discussion tab.

One way to make everything neater is to move the `is current page PR` check into `openPrTab`, like in the original `switchToTab` implementation. (there was a PR url check after the `isCurrentUrlIssue` check either way). I'll do that first.

Regarding the bug, I've added `waitUntilDiscussionPageLoaded` accordingly.